Added documentation for licenses in lakeFS Enterprise to the public docs (#9270)
treeverse/lakeFS: docs/src/enterprise/getstarted/install.md

!!! info
    You can install lakeFS Enterprise without configuring SSO and still benefit from all other lakeFS Enterprise features.

---

### **lakeFS Enterprise License**

??? info "Overview"
    - lakeFS Enterprise requires a valid license to run.
    - The license is a JSON Web Token (JWT) that provides access to the lakeFS Enterprise server and paid features.

>     - A license is tied to a specific installation via an installation ID, and to a specific organization via an organization ID.

Author: @arielshaqed 
My intention was to place this list inside a toggle box labeled **"Overview"** to keep extra details hidden by default, allowing readers to expand the section if they wish to read more (to improve readability). 

**With my change, it currently looks like this:**

<img width="951" height="71" alt="image" src="https://github.com/user-attachments/assets/2ba51c48-6023-4142-8e45-2a51ecbb778e" />

**And after clicking the arrow on the right:**

<img width="974" height="246" alt="image" src="https://github.com/user-attachments/assets/b6fc7069-9c2c-42cc-8a75-4a963f346917" />

**If I remove the tabs before each bullet under the Overview section, it will look like this:**

<img width="987" height="265" alt="image" src="https://github.com/user-attachments/assets/3353f989-dec9-4b9a-b5b7-c0af938c32f8" />

**Which version do you think is better?**

    !!! warning
        You must use exactly one method, providing both or neither will cause the server to fail to start.

??? info "Licensed Features"
    The following lakeFS Enterprise features must be included in the lakeFS Enterprise license in order to be available:

    **Authentication & Authorization**

    - **[SSO (Single Sign-On)](../../security/sso.md)**
    - **[RBAC (Role-Based Access Control)](../../security/rbac.md)**
    - **[SCIM (System for Cross-domain Identity Management)](../../howto/scim.md)**
    - **[IAM (Identity and Access Management) Role Authentication](../../security/external-principals-aws.md)**

    **Advanced Functionality**

    - **[Mount](../../reference/mount.md)**
    - **[Metadata Search](https://info.lakefs.io/metadata-search)**
    - **[MSB (Multiple Storage Backends)](../../howto/multiple-storage-backends.md)**
    - **[Transactional Mirroring](../../howto/mirroring.md)**
    - **[Sparkless GC (Garbage Collection)](../../howto/garbage-collection/standalone-gc.md)**

    !!! warning 

>         Without a license for these features, they will be disabled, and any attempt to access them will result in a 'feature not licensed' error.

Author: Agree, waiting on Oz's input to know how to change this section

>         Without a license for these features, they will be disabled, and any attempt to access them will result in a 'feature not licensed' error.

Reviewer: I think it's good to maintain the list of licensable features here, even if we also have it on the pricing page and other assets. Agree with @N-o-Z that this should come earlier in the doc.

    _When Using File Path:_

    1. Replace the content of your license file with the new license token.  
    2. lakeFS Enterprise will automatically detect and reload the new license within 1 minute.  
    
    !!! warning
        You cannot change the file path itself to point to a new file while the server is running.

    _When Using Direct Contents:_

    1. Update your configuration file with the new license token.
    2. Restart the lakeFS Enterprise server to apply changes.

    **Automatic Monitoring**

    - **Validation and Expiry Check**: Occurs every 24 hours to verify license validity and expiry.
    - **File Monitoring**: When using `license.path`, the server checks every minute to detect license file changes so that when the license token gets updated in the file given in the path, there is no need to restart the server and the license updates automatically.

    **License Expiry Server Behavior**

    - **31+ days before expiry**: No warnings.
    - **30 to 0 days before expiry**: A warning notification appears in the web UI.
    - **1 to 30 days After expiry**: An error notification appears in the web UI and logs, but the server continues running.

>     - **31+ days after expiry**: The server will shut down and refuse to start.

Author: Got it, updated:

```
- **30 to 0 days before expiry:** A warning notification is displayed in the web UI.
- **1 to 30 days After expiry:** An error notification appears, and the server will shut down and refuse to start.
```

Is this ok?

>     - **31+ days after expiry**: The server will shut down and refuse to start.

Author: Removed the **License Expiry Server Behavior** section and replaced it with:

```
!!! warning 
        After the license expires, the server shuts down and fails to start due to a license expiry error.
```

Placed it below the:

`3. **Configure lakeFS Enterprise Server**`

??? info "Reading License Token via API"

    **Request:**
    ```bash
    GET https://your-lakefs-server/api/v1/license
    ```

    **Response:**  
    ```json
    {
      "token": "eyJhbGciOiJSUzI1NiIs..."
    }

>     ```

Reviewer 2: Maybe note that the token can be decoded by any JWT decoding tool to see the license information

??? question "Troubleshooting"
    - **Server Won't Start**
    ```
    Error: no valid license found for this lakeFS Enterprise Server
    ```

>     **Solution:** Ensure you have configured either `license.path` or `license.contents` in the lakeFS Enterprise configuration file or environment variables.

Reviewer 2: Remove reference to contents

    ```
    Error: cannot provide both license path and contents
    ```

>     **Solution:** Remove one of the license configuration methods from your lakeFS Enterprise configuration file or environment variables.

Reviewer 2: I would remove that

    ```
    Error: license has expired. Please contact support immediately: invalid license found
    ```
    ```
    Error: license expired: invalid expiry date
    ```
    **Solutions:**  
    &nbsp;1. If recently renewed, ensure you've updated the license file with the new token.  
    &nbsp;2. Contact [support@treeverse.io](mailto:support@treeverse.io) immediately for license renewal.  

    ---

    - **License File Not Found**
    ```
    Error: open /path/to/license/file/you/provided: no such file or directory
    ```
    **Solutions:**  
    &nbsp;1. Verify the exact path and filename in your `license.path` configuration.  
    &nbsp;2. Check for typos in filename.  
    &nbsp;3. Check that license file wasn't moved or removed after configuration.  
    &nbsp;4. Check that the license file exists at the specified location.  
    &nbsp;5. Ensure the file extension is included.  
    &nbsp;6. Use absolute paths instead of relative paths.  

    ---

    - **License File Permission Denied**
    ```
    Error: open /path/to/license/file/you/provided: permission denied
    ```
    **Solutions:**  
    &nbsp;1. Check file permissions and set appropriate permissions.  

>     &nbsp;2. Ensure lakeFS Enterprise user can read the file and access parent directories.  

Reviewer 2: ```suggestion
    &nbsp;2. Ensure lakeFS process has permission to read the file and access parent directories.  
```

    ---

    - **Installation ID Mismatch**
    ```
    Error: license belongs to installation ID X (current installation ID: Y)
    ```
    **Solutions:**  
    &nbsp;1. Check if you have the correct license file for this specific installation.  
    &nbsp;2. Check that your installation ID didn't change.  
    &nbsp;3. Contact [support@treeverse.io](mailto:support@treeverse.io) to get a correct new license.  

    ---

    - **Malformed or Empty License Token**
    ```
    Error: parsing token: token is malformed: token contains an invalid number of segments
    ```
    ```
    Error: license has invalid expiry (no license?): invalid license found
    ```
    **Solutions:**  

>     &nbsp;1. Verify the license file contains the complete license token (should start with `eyJ`).  

Reviewer 2: Might be true currently but I don't see a reason to add this to the docs (should start with `eyJ`)

    &nbsp;2. Remove any extra whitespace, newlines, or characters from the license file.  
    &nbsp;3. Re-copy the license token from the original source.  
    &nbsp;4. Ensure the file contains only the license token and nothing else.  

    ---

    - **Feature Not Available**
    ```
    Error: feature not licensed - to enable, contact support@treeverse.io
    ```
    **Solution:** Your current license doesn't include this feature. Contact [support@treeverse.io](mailto:support@treeverse.io) to upgrade your license.

> 

Reviewer 2: Worth adding a last note for any other issue contact support (link)
